### PR TITLE
feat(replays): Use user configured PII scrubbers

### PR DESCRIPTION
**Pull request by cmanallen**
Removes statically defined scrubbers with data scrubbing and PII config from the project state. This includes a refactor that moves replay scrubbing on a struct.

closes: https://github.com/getsentry/replay-backend/issues/243

### CODE REVIEW

**Comment by cmanallen** [comment]
@JoshFerge

**Comment by jan-auer** [comment]
This will have to be updated to #1800 and requires a bit of cleanup in the `StringTranscoder`. I'll look into this.

File: relay-replays/src/recording.rs [hunk 1]
```diff
@@ -43,104 +41,152 @@ impl From<serde_json::Error> for ParseRecordingError {
...
+    use relay_general::pii::PiiConfig;
+    use relay_general::processor::SelectorSpec;
+
+    use super::ReplayScrubber;
+
+    fn default_pii_config() -> PiiConfig {
+        let mut pii_config = PiiConfig::default();
+        pii_config.applications = [(SelectorSpec::And(vec![]), vec!["@common".to_string()])].into();
+        pii_config
+    }
```
Comment (jjbayer) [inline relay-replays/src/recording.rs]: To have the tests resemble the default production use case, I would use a configuration derived from `DataScrubbingConfig` here, like for example https://github.com/getsentry/relay/blob/afa6576707a77c45ed61b277f7900a0e2f835ba5/relay-general/src/protocol/replay.rs#L554-L560

**Review by jjbayer — CHANGES_REQUESTED** [review]
The implementation looks good to me, just one remark about the test config.

File: relay-replays/src/recording.rs (hunk 1 above)
Comment (jan-auer) [reply relay-replays/src/recording.rs]: Thanks, updated. This actually shows us that the filter has changed.

@cmanallen note that email addresses aren't being scrubbed anymore, apparently the default data scrubbers don't do this. No action item imho, this is in line with the other data scrubbers.

**Review by jjbayer — APPROVED** [review]

**Comment by cmanallen** [comment]
@jjbayer @jan-auer Should this be merged?

**Comment by jan-auer** [comment]
Yes. I was waiting for a response on https://github.com/getsentry/relay/pull/1731#discussion_r1094548564. 

Merging now.

**Comment by jan-auer** [comment]
Yes. I was waiting for a response on https://github.com/getsentry/relay/pull/1731#discussion_r1094548564. 

Merging now.

**Pull request merged**